include/safe/int.hpp

```diff
@@ -58,16 +58,43 @@ template <auto min, auto max> using ival_u64 = var<uint64_t, ival<min, max>>;
 using namespace interval_types;
 
 namespace detail {
+constexpr static auto is_delimiter = [](char const c) -> bool {
+    return c == '\'';
+};
+
 template <typename T, char... Chars>
+    requires decimal_integer<T, Chars...>
 [[nodiscard]] constexpr auto to_constant() {
-    // FIXME: handle or fail at compile-time for invalid strings
     constexpr T value = []() {
         constexpr std::array<char, sizeof...(Chars)> chars{Chars...};
         T sum = 0;
 
         for (char c : chars) {
-            T const digit = c - '0';
-            sum = (sum * 10) + digit;
+            if (not is_delimiter(c)) {
+                T const digit = c - '0';
+                sum = (sum * 10) + digit;
+            }
+        }
+
+        return sum;
+    }();
+
+    return make_constant<T, value>();
+}
+
+template <typename T, char Char0, char Char1, char... Chars>
+    requires hex_integer<T, Char0, Char1, Chars...>
+[[nodiscard]] static constexpr auto to_constant() {
+    constexpr T value = []() {
+        constexpr std::array<char, sizeof...(Chars)> chars{Chars...};
+        T sum = 0;
+
+        for (char c : chars) {
+            if (not is_delimiter(c)) {
+                T const digit =
+                    c > '9' ? c >= 'a' ? c - 'a' + 10 : c - 'A' + 10 : c - '0';
```

Comment:
**suggestion**: Instead of nested ternary operator, how do you feel about e.g.
`T const digit = c <= '9' ? c - '0' : (c & 0b1101'1111) - 'A' + 10;`

Reply:
I see, the difference between upper and lower case in ascii table is the presence of the 6th bit. Didn't know about it. I like it this nerdy approach! Will leave an explanatory comment if somebody like me will stuble